storages/python/python-storage-plugins/bson.py

```diff
@@ -0,0 +1,170 @@
+"""A DLite storage plugin for BSON written in Python."""
+import os
+import subprocess
+import sys
+
+import binascii
+from binascii import unhexlify
+import bson as pybson # Must be pymongo.bson
+
+import dlite
+from dlite.options import Options
+from dlite.utils import instance_from_dict
+
+
+class bson(DLiteStorageBase):
+    """DLite storage plugin for BSON."""
+
+    def open(self, uri, options=None):
+        """Open `uri`.
+
+        The `options` argument provies additional input to the driver.
+        Which options that are supported varies between the plugins.
+        It should be a valid URL query string of the form:
+
+            key1=value1;key2=value2...
+
+        An ampersand (&) may be used instead of the semicolon (;).
+
+        Typical options supported by most drivers include:
+        - mode : append (default) | r | w
+            Valid values are:
+            - append   Append to existing file or create new file
+            - r        Open existing file for read-only
+            - w        Truncate existing file or create new file
+
+        After the options are passed, this method may set attribute
+        `writable` to True if it is writable and to False otherwise.
+        If `writable` is not set, it is assumed to be True.
+        
+        The BSON data is translated to JSON.
+        """
+        self.options = Options(options, defaults='mode=append')
+        self.mode = dict(r='rb', w='wb', append='rb+')[self.options.mode]
+        self.writable = False if 'rb' in self.mode else True
+        self.uri = uri
+        self.d = {}
+        if self.mode in ('rb', 'rb+') and os.path.exists(uri):
```

Comment:
This will not fail if `uri` doesn't exists - but we want it to fail in this case since we are opening in read mode

Reply:
Ok. Fail, as in raising an exception?

Reply:
Yes, raising an exception with a good message would be the preferred behaviour